L1Trigger/L1THGCal/plugins/backend/HGCalBackendLayer2Processor3DClustering_SA.cc

```diff
@@ -41,35 +44,61 @@ class HGCalBackendLayer2Processor3DClusteringSA : public HGCalBackendLayer2Proce
     l1t::HGCalClusterBxCollection& rejectedClusters = be_output.second;
 
     /* create a persistent vector of pointers to the trigger-cells */
-    std::vector<edm::Ptr<l1t::HGCalCluster>> clustersPtrs;
+    std::unordered_map<uint32_t, std::vector<edm::Ptr<l1t::HGCalCluster>>> tcs_per_fpga;
+
     for (unsigned i = 0; i < collHandle->size(); ++i) {
-      edm::Ptr<l1t::HGCalCluster> ptr(collHandle, i);
-      clustersPtrs.push_back(ptr);
-    }
+      edm::Ptr<l1t::HGCalCluster> tc_ptr(collHandle, i);
+      uint32_t module = geometry_->getModuleFromTriggerCell(tc_ptr->detId());
+      uint32_t stage1_fpga = geometry_->getStage1FpgaFromModule(module);
+      HGCalTriggerGeometryBase::geom_set possible_stage2_fpgas = geometry_->getStage2FpgasFromStage1Fpga(module);
```

Comment:
I think `getStage2FpgasFromStage1Fpga` should take `stage1_fpga` as input instead of `module`

Reply:
thanks, changed